packages/gatsby/src/utils/api-node-docs.js

```diff
@@ -12,49 +12,51 @@ exports.resolvableExtensions = true
  *
  * See also [the documentation for the action `createPage`](/docs/actions/#createPage).
  * @example
+ * const path = require(`path`)
+ *
  * exports.createPages = ({ graphql, actions }) => {
  *   const { createPage } = actions
- *   return new Promise((resolve, reject) => {
- *     const blogPostTemplate = path.resolve(`src/templates/blog-post.js`)
- *     // Query for markdown nodes to use in creating pages.
- *     resolve(
- *       graphql(
- *         `
- *       {
- *         allMarkdownRemark(limit: 1000) {
- *           edges {
- *             node {
- *               fields {
- *                 slug
- *               }
+ *   const blogPostTemplate = path.resolve(`src/templates/blog-post.js`)
+ *   // Query for markdown nodes to use in creating pages.
+ *   // You can query for whatever data you want to create pages for e.g.
+ *   // products, portfolio items, landing pages, etc.
+ *   graphql(`
```

Comment:
Do we want to return here? It doesn't really matter either way, but it doesn't _hurt_